Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Beckn/Select.hs

```diff
@@ -159,9 +160,53 @@ buildSearchReqLocation merchantId sessionToken _ latLong@Maps.LatLong {..} = do
           sessionToken = Just sessionToken,
           language = Nothing
         }
-  Address {..} <- mkLocation pickupRes
-  id <- Id <$> generateGUID
-  now <- getCurrentTime
-  let createdAt = now
-      updatedAt = now
-  pure DLoc.SearchReqLocation {..}
+  add <- mkLocation pickupRes
+  if (DT.length <$> add.area) /= Just 0
+    then do
+      id <- Id <$> generateGUID
+      now <- getCurrentTime
+      let createdAt = now
+          updatedAt = now
+      pure
+        DLoc.SearchReqLocation
+          { id = id,
+            lat = latLong.lat,
+            lon = latLong.lon,
+            street = add.street,
+            city = add.city,
+            state = add.state,
+            country = add.country,
+            building = add.building,
+            areaCode = add.areaCode,
+            area = add.area,
+            full_address = add.full_address,
+            createdAt = createdAt,
+            updatedAt = updatedAt
+          }
+    else do
+      getNearBySearchResp <-
+        Maps.getNearBySearch merchantId $
+          Maps.GetNearBySearchReq
+            { location = getLocation latLong,
+              rankby = "distance"
+            }
+      let Address {..} =
+            add{area = Just getNearBySearchResp,
+                full_address = getfullAddress (Just getNearBySearchResp) city state areaCode country
+               }
+      id <- Id <$> generateGUID
+      now <- getCurrentTime
+      let createdAt = now
+          updatedAt = now
+      pure DLoc.SearchReqLocation {..}
+
+getLocation :: LatLong -> Text
+getLocation latlong = show latlong.lat <> "," <> show latlong.lon
+
+getfullAddress :: Maybe Text -> Maybe Text -> Maybe Text -> Maybe Text -> Maybe Text -> Maybe Text
+getfullAddress area city state areaCode country = do
```

Comment:
areaCode is not used
can you remove {-# OPTIONS_GHC -Wno-unused-matches #-} pragma for more safety?

Reply:
areaCode is NULL in those cases